### ISSUE TITLE
Float values for backoff.factor?

### ISSUE THREAD

**Issue by max-sixty** [argoproj/argo-workflows#6454]
# Summary

Currently supplying a float value for `retryStrategy.backoff.factor` will raise a JSON unmarshalling error. 

Any thoughts on allowing float values?

# Use Cases

Mandating ints gives a minimum of a factor of 2, which leads to very long backoff times with many retries — i.e. IIUC the final attempt of 10 retries with a duration of 1 minute will be 31 hours after the first.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by terrytangyuan** [comment]
I agree this should be a float instead. We currently support env var RETRY_BACKOFF_DURATION which is float so we should probably be consistent there.

**Comment by weafscast** [comment]
@terrytangyuan I can take this up, just wanted to ask if I should make this FloatOrStr or if it could be float.
